tests/integration/istioctl/istioctl_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"bytes"
+	"regexp"
+	"strings"
+	"testing"
+
+	"istio.io/istio/istioctl/cmd"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var (
+	i   istio.Instance
+	env *kube.Environment
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("istioctl_integration_test", m).
+		Label(label.Presubmit).
+
+		// The following is how to deploy Istio on Kubernetes, as part of the suite setup.
+		// The deployment must work. If you're breaking this, you'll break many integration tests.
+		SetupOnEnv(environment.Kube, istio.Setup(&i, nil)).
+		SetupOnEnv(environment.Kube, func(ctx resource.Context) error {
+			env = ctx.Environment().(*kube.Environment)
+			return nil
+		}).
+
+		// Finally execute the test suite
+		Run()
+}
+
+// TestVersion does "istioctl version --remote=true" to verify the CLI understands the data plane version data
+func TestVersion(t *testing.T) {
+	framework.Run(t, func(ctx framework.TestContext) {
+		args := []string{"version", "--remote=true"}
```

Comment:
You don't necessarily need the test framework for this particular test. I am assuming you'll be writing more tests with it?

Reply:
I plan to write more tests once I get the first one right.

We already have a Go unit test that does this with `--remote=false` and one that does it with `--remote=true` and mocking of remote results.  This test is intended to test the communication for real.  It was inspired by how the output of `pilot-discovery version` changed between Istio 1.0 and 1.1.  If `pilot-discovery version` changes again I want to test so that we can change istioctl at the same time to handle the new format.

However, to really test that case the test needs to find some Istio pods.  Currently it doesn't.  Am I supposed to add flags so the framework starts a data plane?  Add code to wait for the data plane to reach Running state?

Current output:
```
2019-05-07T15:41:56.144614Z	info	tf	Test run dir: /tmp/istioctl-integration-test-13dca
--- FAIL: TestVersion (0.01s)
    istioctl_test.go:67: Unwanted exception for 'istioctl version --remote=true': unable to find any Istio pod in namespace istio-system
```